include/amgx_config.h

```diff
@@ -98,107 +98,11 @@ typedef enum
-    AMGX_mode_hIDI = AMGX_ASSEMBLE_MODE(AMGX_host, AMGX_vecInt,  AMGX_matDouble,  AMGX_indInt),
-    AMGX_mode_hIFI = AMGX_ASSEMBLE_MODE(AMGX_host, AMGX_vecInt,  AMGX_matFloat,  AMGX_indInt),
-    AMGX_mode_dIDI = AMGX_ASSEMBLE_MODE(AMGX_device, AMGX_vecInt,  AMGX_matDouble,  AMGX_indInt),
-    AMGX_mode_dIFI = AMGX_ASSEMBLE_MODE(AMGX_device, AMGX_vecInt,  AMGX_matFloat,  AMGX_indInt),
-    AMGX_mode_hZZI = AMGX_ASSEMBLE_MODE(AMGX_host,   AMGX_vecDoubleComplex,  AMGX_matDoubleComplex, AMGX_indInt), // mode == 8192
-    AMGX_mode_hZCI = AMGX_ASSEMBLE_MODE(AMGX_host,   AMGX_vecDoubleComplex,  AMGX_matComplex,  AMGX_indInt), // mode == 8448
-    AMGX_mode_hCCI = AMGX_ASSEMBLE_MODE(AMGX_host,   AMGX_vecComplex,        AMGX_matComplex,  AMGX_indInt), // mode == 8464
-    AMGX_mode_dZZI = AMGX_ASSEMBLE_MODE(AMGX_device, AMGX_vecDoubleComplex,  AMGX_matDoubleComplex, AMGX_indInt), // mode == 8193
-    AMGX_mode_dZCI = AMGX_ASSEMBLE_MODE(AMGX_device, AMGX_vecDoubleComplex,  AMGX_matComplex,  AMGX_indInt), // mode == 8449
-    AMGX_mode_dCCI = AMGX_ASSEMBLE_MODE(AMGX_device, AMGX_vecComplex,        AMGX_matComplex,  AMGX_indInt), // mode == 8465
-    AMGX_modeNum = 10,
-    AMGX_ModeInst
-} AMGX_Mode;
-
-/* Builds for host */
-#ifdef AMGX_build_host
-#define AMGX_FORALL_BUILDS_HOST(codeLineMacro)\
-  codeLineMacro(AMGX_build_host)
-#else
-#define AMGX_FORALL_BUILDS_HOST(codeLineMacro)\
-  codeLineMacro(AMGX_mode_hDDI)\
-  codeLineMacro(AMGX_mode_hDFI)\
-  codeLineMacro(AMGX_mode_hFFI)
-#endif
-
-#define AMGX_FORINTVEC_BUILDS_HOST(codeLineMacro)\
-  codeLineMacro(AMGX_mode_hIDI)\
-  codeLineMacro(AMGX_mode_hIFI)
-
-/* Builds for device */
-#ifdef AMGX_build_device
-#define AMGX_FORALL_BUILDS_DEVICE(codeLineMacro)\
-  codeLineMacro(AMGX_build_device)
-#else
-#define AMGX_FORALL_BUILDS_DEVICE(codeLineMacro)\
-  codeLineMacro(AMGX_mode_dDDI)\
-  codeLineMacro(AMGX_mode_dDFI)\
-  codeLineMacro(AMGX_mode_dFFI)
-#endif
-
-#define AMGX_FORINTVEC_BUILDS_DEVICE(codeLineMacro)\
-  codeLineMacro(AMGX_mode_dIDI)\
-  codeLineMacro(AMGX_mode_dIFI)
-
-#define AMGX_FORCOMPLEX_BUILDS_DEVICE(codeLineMacro)\
-  codeLineMacro(AMGX_mode_dZZI)\
-  codeLineMacro(AMGX_mode_dZCI)\
-  codeLineMacro(AMGX_mode_dCCI)
-
-#define AMGX_FORCOMPLEX_BUILDS_HOST(codeLineMacro)\
-  codeLineMacro(AMGX_mode_hZZI)\
-  codeLineMacro(AMGX_mode_hZCI)\
-  codeLineMacro(AMGX_mode_hCCI)
-
-/* Builds for device and host */
-#define AMGX_FORALL_BUILDS(codeLineMacro)\
-  AMGX_FORALL_BUILDS_HOST(codeLineMacro)\
-  AMGX_FORALL_BUILDS_DEVICE(codeLineMacro)
-
-#define AMGX_FORCOMPLEX_BUILDS(codeLineMacro)\
-  AMGX_FORCOMPLEX_BUILDS_DEVICE(codeLineMacro)\
-  AMGX_FORCOMPLEX_BUILDS_HOST(codeLineMacro)
-
-#define AMGX_FORINTVEC_BUILDS(codeLineMacro)\
-  AMGX_FORINTVEC_BUILDS_HOST(codeLineMacro)\
-  AMGX_FORINTVEC_BUILDS_DEVICE(codeLineMacro)
-
+// User-defined types
+constexpr AMGX_MemorySpace AMGX_MemSpace = AMGX_device;
```

Comment:
Hey Matthew. Is idea here to compile with one set of precisions only? If so - would it make sense to make parameter to CMake? (i.e. `cmake -DMAT_PRECISION=Double -DIND_PRECISION=Int64 ...`)